### PR TITLE
Unify logging in tests/*

### PR DESCRIPTION
**Pull request by aquesnel**
Follow-on pull request to #1633 which migrates all logging in the tests directory to use the LOG() and LOG_DEVEL() macros.

Changes:

* Migrate logging to the LOG() and LOG_DEVEL() macros (using text transform script)
* error logs use the LOG macro
* update code formatting with astyle for all files in tests/*
* add building the tests directory when running `make all` and `make check`
* refactor the hex dump code to be a string function rather than a logging function
* add `LOG_HEXDUMP` for always logging a hexdump

### CODE REVIEW

**Review by matt335672 — COMMENTED** [review]
Great work, as ever.

**Comment by matt335672** [comment]
While reviewing this I was thinking about [this recent comment](/neutrinolabs/xrdp/pull/1818#issuecomment-789559829) which touches on this.

I'm wondering about the structure of the tests directory now, in the context of this PR.

Now that `make check` does something useful and expected with the `tests` directory, should we move `tests/tcp_proxy/` and `tests/gtcp_proxy/` elsewhere?

Any thoughts on this @metalefty?

**Comment by aquesnel** [comment]
@matt335672 I agree that `tcp_proxy` and `gtcp_proxy` should be moved out of the tests directory. My suggestions for new top level directory names are:
* devel_tools
* devutils
* utils
What do you think?

File: common/string_calls.h [hunk 1]
```diff
@@ -97,6 +97,7 @@ int      g_atoi(const char *str);
 int      g_htoi(char *str);
 int      g_bytes_to_hexstr(const void *bytes, int num_bytes, char *out_str,
                            int bytes_out_str);
+char    *g_bytes_to_hexdump(const char *src, int len);
```
[the remark this answers is not in the document]
Comment (aquesnel) [reply common/string_calls.h]: Added documentation.

**Comment by aquesnel** [comment]
@matt335672  I've moved the tcp_proxy programs to the utils directory, and I've added docs for g_bytes_to_hexdump. Please let me know if there is any other feedback for this change to be ready to merge.

**Comment by Nexarian** [comment]
In general I don't recommend using catch-all terms like `utils` -- It often begs the question "utils for what?" And then becomes a dumping ground for all sorts of code that is unrelated, thus making it harder to maintain.

**Comment by metalefty** [comment]
I personally never used [g]tcp_proxy tool. `socat` can do the same thing as the tools. We can throw them out to a separate repository.

**Comment by metalefty** [comment]
Ah, the difference between tcp_proxy and socat is traffic dumping. It looks like a developer's tool to debug, not a user's tool.
As @Nexarian mentioned, the name `utils` is not clear. What about a name like `devtool` instead?

File: common/string_calls.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (matt335672) [reply common/string_calls.h]: Works for me!

**Comment by aquesnel** [comment]
I've made the parent directory for the tcp_proxy tools more specific, by moving it to `tools/devel`. I've chosen `devel` instead of `dev` since the rest of the code uses the abbreviation `devel` for development/developer.

**Comment by metalefty** [comment]
I'm busy at daytime job. With a quick look thru, I'm fine with this.
Shit it if it looks good to @matt335672.

**Comment by metalefty** [comment]
s/Shit/Ship/  😉

File: common/string_calls.h [hunk 2]
```diff
@@ -80,6 +80,29 @@ g_bool2text(int value);
...
+/**
+ * Converts a binary array into a hux dump suitable for displaying to a user.
+ *
+ * The format of the hex dump is:
+ * 0000   01 02 03 04 05 06 07 08 09 10 11 12 13 14 15 16   ................
+ * /\     /\                                                /\
+ * |      |                                                 |
+ * |      |                                                 ascii representation of bytes
+ * |      hex representation of bytes
+ * offset from beining of the byte array in hex
```
Comment (matt335672) [inline common/string_calls.h:92]: beining -> beginning

**Comment by matt335672** [comment]
Sorry about missing the typo above earlier.

@aquesnel - please can you fix the typo rebase this and squash it ? Then I'll merge.

Thanks.

File: common/string_calls.h (hunk 2 above)
Comment (aquesnel) [reply common/string_calls.h:92]: fixed

**Comment by aquesnel** [comment]
@matt335672 the spelling mistake is fixed, and the branch is already rebased on top of devel.

The commits in this PR are all self contained logical commits, so I'd recommend not squashing them together before the merge. If you still think that the commits should be squashed, then I'll squash and update the PR.

**Comment by matt335672** [comment]
No - that's fine.

Thanks again for this one, particularly making the tests usable.

**Pull request merged**